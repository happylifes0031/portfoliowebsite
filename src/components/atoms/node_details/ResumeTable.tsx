import React from "react";
import styled from "@emotion/styled";

interface ResumeTableProps {
  from: string;
  till: string;
  role: string;
  skills: string;
}

const StyledResumeTable = styled(`div`)`
  position: absolute;
  width: 220px;
  background-color: orange;
  left: 23px;
  top: 71px;
  font-size: 0.7rem;
  padding: 1px;
  border-radius: 5px;
  z-index: 999;
  animation: fadein 1.5s;
  display: "block";
  ul {
    display: flex;
    flex-wrap: wrap;
    background-color: #efefef;
    list-style: none;
    padding: 0;
    margin: 0;
    text-align: left;
    li {
      padding: 6px 0 3px 2px;
      border-bottom: 1px black dashed;
      &:last-of-type,
      &:nth-child(5) {
        border-bottom: none;
      }
      &:nth-child(odd) {
        width: 30%;
      }
      &:nth-child(even) {
        width: 68%;
      }
    }
  }
`;

const ResumeTable = ({ from, till, role, skills}: ResumeTableProps) => (
  <StyledResumeTable>
    <ul>
      <li>
        <b>Timetable:</b>
      </li>
      <li>
        {from} | {till}
      </li>
      <li>
        <b>Rol:</b>
      </li>
      <li>{role}</li>
      <li>
        <b>Skills:</b>
      </li>
      <li>{skills}</li>
    </ul>
  </StyledResumeTable>
);

export default ResumeTable;
