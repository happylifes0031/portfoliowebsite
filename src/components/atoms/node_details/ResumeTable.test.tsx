import React from "react";
import renderer from "react-test-renderer";
import { matchers } from "jest-emotion";
import "jest-styled-components";

import ResumeTable from "./ResumeTable";
expect.extend(matchers);

describe("ResumeTable", () => {
  it("Renders as expected", () => {
    const resumeTable = renderer
      .create(<ResumeTable from={''} till={''} role={''} skills={''} />)
      .toJSON();
    expect(resumeTable).toMatchSnapshot();
  });
});
