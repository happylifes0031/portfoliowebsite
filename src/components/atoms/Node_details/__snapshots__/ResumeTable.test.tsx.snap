// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResumeTable Renders as expected 1`] = `
<div
  className="css-8pl6jd"
>
  <ul>
    <li>
      <b>
        Timetable:
      </b>
    </li>
    <li>
      
       | 
      
    </li>
    <li>
      <b>
        Rol:
      </b>
    </li>
    <li>
      
    </li>
    <li>
      <b>
        Skills:
      </b>
    </li>
    <li>
      
    </li>
  </ul>
</div>
`;
